lib/generator.js

```diff
@@ -1,38 +1,159 @@
 'use strict';
 
+const common = require('metarhia-common');
+
+const structures = [
+  'boolean',
+  'null',
+  'undefined',
+  'number',
+  'string',
+  'symbol',
+];
+
+const gObjects = [
+  'Object',
+  'Array',
+  'Date',
+  'Function',
+  'Map',
+  'WeakMap',
+  'Set',
+  'WeakSet',
+  'Int8Array',
+  'Uint8Array',
+  'Uint8ClampedArray',
+  'Float32Array',
+  'Error',
+  'EvalError',
+  'TypeError',
+  'RangeError',
+  'SyntaxError',
+  'ReferenceError',
+];
+
+const types = [
+  ...structures,
+  ...gObjects,
+  'Primitive',
+];
+
+const baseUrl = 'https://developer.mozilla.org/en-US/docs/';
+const structuresBaseUrl = 'Web/JavaScript/Data_structures#';
+const gObjectsBaseUrl = 'Web/JavaScript/Reference/Global_Objects/';
+
+const structuresLinks = structures.map(type =>
+  [type, `${baseUrl}${structuresBaseUrl}${common.capitalize(type)}_type`]
+);
+
+const gObjectsLinks = gObjects.map(type =>
+  [type, `${baseUrl}${gObjectsBaseUrl}${type}`]
+);
+
+let links = [
+  ...gObjectsLinks,
+  ...structuresLinks,
+  [ 'Primitive', baseUrl + 'Glossary/Primitive' ],
+];
+
+links = links.map(link => '[`<' + link[0] + '>`]: ' + link[1]);
+
+const wrapType = type => {
+  if (types.includes(type)) {
+    return '[`<' + type + '>`][]';
+  } else if (type.startsWith('[')) {
+    type = type.slice(1, type.length - 1)
+      .replace(/ /, '')
+      .split(',')
+      .map(s => s.slice(1, s.length - 1))
+      .map(wrapType)
+      .join('`, `');
+    return '`[ `' + type + '` ]`';
+  } else {
+    return '`<' + type + '>`';
+  }
+};
+
+const generateParameters = parameters => {
+  const buf = [];
+  for (const parameter of parameters) {
+    if (Array.isArray(parameter.type)) {
+      parameter.type = parameter.type.map(wrapType).join('` | `');
+    } else if (parameter.type) {
+      parameter.type = wrapType(parameter.type);
+    }
+    if (parameter.comment) {
+      if (parameter.nonStandartTypes) {
+        parameter.comment =
+          parameter.nonStandartTypes.map(wrapType).join('` | `') +
+          parameter.comment;
+      }
+      parameter.comment = ' - ' + parameter.comment;
+    }
+    const pad = ' '.repeat(parameter.offset);
+    buf.push(
+      (pad + '- `' + parameter.name + (parameter.type ? ': `' : '`') +
+        parameter.type + parameter.comment).replace(/``/g, '')
+    );
+  }
+  return buf;
+};
+
+const generateRest = comments => {
+  const buf = [];
+  for (const comment of comments) {
+    if (comment.name === 'Example' || comment.name === 'Result') {
+      comment.comment = '\n```js\n' + comment.comment + '\n```';
```

Comment:
Could we rename nested `comment` to smth like `source`? (i.e. `comment.source`) IIUC